lib/x11utils.pm

```diff
@@ -68,6 +68,17 @@ all possible options should be handled within loop to get unlocked desktop
 =cut
 sub ensure_unlocked_desktop {
     my $counter = 10;
+
+    # check if the screenbuffer was not updated and record a softfailure in that case
+    if (check_var('DESKTOP', 'gnome') && check_screen('screenlock')) {
+        send_key 'shift';
+        wait_still_screen;    # redrawing, turning on guest display or showing animation may take some seconds
```

Comment:
please do not do that. `wait_still_screen` without explicit still time value is a big time waste which you call here without checking for any error condition in before. Still my old comment still stays and I did not see a response (sorry if I missed it): You are adding another layer of complexity by doing some checks outside the while loop for an issue that is already addressed in a slightly different way within the following while-loop as well.

Reply:
>  my old comment still stays and I did not see a response (sorry if I missed it)

My first suggestion was to only include the workaround in the general-desktop if-branch, but:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10064#discussion_r413838590

> You are adding another layer of complexity by doing some checks outside the while loop for an issue that is already addressed in a slightly different way within the following while-loop as well.

There are 5 if-branches. Only one of them rules out the bug (screenlock), the other four could reproduce the bug. That means that the check outside of the loop would need to be done inside each of those 4 if-branches, duplicating code.

Reply:
I see. Well, let's try it your way then :)

Reply:
Thought about this again and have commented in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10064#issuecomment-625483397 so I am staying with my original assessment: Please do not add the additional logic outside the loop and no wait_still_screen with default stilltime.

Reply:
I'm siding with @okurz here. As I mentioned to @DrMullings earlier today: Initial suggestions to this ticket, were wrong.